### PR TITLE
Add new tests for Swagger.ObjectModel

### PR DESCRIPTION
**Pull request by catcherwong**
For this PR , I have done the tests for the following builder classes:

- [x]  InfoBuilder 
- [x]  LicenseBuilder
- [x]  OperationBuilder
- [x]  ParameterBuilder
- [x]  PathItemBuilder
- [x]  ResponseBuilder
- [x]  SecurityRequirementBuilder
- [x] SwaggerRootBuilder
- [x] TagBuilder 

And there are for builder classes need to add some tests  

- [ ] ApiKeySecuritySchemeBuilder
- [ ] BasicSecuritySchemeBuilder
- [ ] Oauth2SecuritySchemeBuilder
- [ ] SchemaBuilder

### CODE REVIEW

File: test/Swagger.ObjectModel.Tests/Builders/InfoBuilderTest.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        [InlineData("title", null)]
+        public void Should_ThrowRequiredFieldException_WhenTitleOrVersionIsEmptyOrWhiteSpace(string title, string version)
+        {
+            Assert.Throws(typeof(RequiredFieldException), () => new InfoBuilder(title, version).Build());
+        }
+
+        [Theory]
+        [InlineData("title", "1.0","description")]
+        [InlineData("title", "1.0", "")]
+        public void Should_AbleToSetDescription(string title, string version,string descrption)
```
Comment (jnallard) [inline test/Swagger.ObjectModel.Tests/Builders/InfoBuilderTest.cs]: Minor formatting: 
You forgot a space: `version,string` 
You also missed one two lines above: `"1.0","description"`

You should double check the rest of the files too.

**Comment by jnallard** [comment]
This looks fine too me. 
Some tests seems good, such as throwing exceptions for required fields. 
I'm not sure how helpful some of the tests are, such as the ones that test that basic assignment works, but I don't see the harm either. 

I'll leave it up to @yahehe to approve and merge.

File: test/Swagger.ObjectModel.Tests/Builders/InfoBuilderTest.cs (hunk 1 above)
Comment (yahehe) [reply test/Swagger.ObjectModel.Tests/Builders/InfoBuilderTest.cs]: It's probably worth us integrating a linter in our build process for style things like this, I expect there will be lots of minor fixes across the whole repo if we do get that up and running so I'm fine considering this non-blocking (though I would prefer if it gets fixed first!)

**Comment by yahehe** [comment]
This looks fantastic, thank you very much @catcherwong !

**Comment by yahehe** [comment]
I'll take care of any formatting-type errors when I add stylecop

**Comment by catcherwong** [comment]
@jnallard  Based on your comment , maybe I should stop adding unit test for **Swagger.ObjectModel** and turn to **Nancy.Swagger** at first ! :smile: :smile:

**Comment by jnallard** [comment]
@catcherwong Nah, it's fine! Adding tests everywhere is good (and you're exhibiting more patience than I have)

**Comment by yahehe** [comment]
If it helps, I think even the most basic tests are useful, even if they're just testing assignment. Tests aren't just for verifying that the code does what it's supposed to, but also that behavior doesn't change in subtle ways (especially easy with C# properties)

Thanks for your contributions so far Catcher, hopefully looking forward to more :)

**Pull request merged**